### DISCUSSION TITLE
Previous prompt broken?

### DISCUSSION THREAD

**Discussion by Holt59** [JanDeDobbeleer/oh-my-posh#2678] [Troubleshoot] [answered]
I've tried upgrading from the old oh-my-posh to a new one and I'm getting a weird issue where, after executing a command, the last prompt revert to a `pwsh>`

![image](https://user-images.githubusercontent.com/2393288/185207175-b9f92fb4-570c-4466-b6e3-be3f7718409e.png)

I'm using the exact same profile and theme on another computer (where I started directly from the new oh-my-posh) and it works fine.

**Comment by JanDeDobbeleer** [comment, marked as answer]
This is because of the transient prompt being enabled. Not sure why if there's no config entry, but that's it.

**Reply by Holt59** [reply]
Thanks, that was the issue. I did have a `transient_prompt` configuration on both machine (I don't know why, probably a copy/paste from a template configuration), it was not shown on the other computer because `oh-my-posh` was not up-to-date. Updating `oh-my-posh` on the "working" computer duplicates the behavior, so I can simply remove the transient prompt configuration.

**Comment by lewis-yeung** [comment]
> I'm using the exact same profile and theme on another computer (where I started directly from the new oh-my-posh) and it works fine.

@Holt59 The exact same profile? Not sure if my thoughts help...

If you have two init lines (`oh-my-posh init pwsh ...`) in your profile, it can cause a problem where the transient prompt is enabled even if in the second init line you have specified a theme that does not contain a `transient_prompt` config. I mentioned the case [in another comment](https://github.com/JanDeDobbeleer/oh-my-posh/pull/2643#issuecomment-1210110815):

> For the first `oh-my-posh init ...` command that has no config file specified, by default, Oh My Posh uses [a "default" theme that contains a `transient_prompt` config](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/default.omp.json#L184-L188) so that the initialization script enables the transient prompt feature in the current PowerShell session. Once enabled, even though a second `oh-my-posh init ...` command uses a config without a `transient_prompt` section, the behaviors cannot be reverted.

Another similar possibility: You set the environment variable `POSH_THEME` to your config path after the only init line (without a config specified), which may bring the same results.
